lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -324,14 +324,30 @@ namespace Js
     typedef FinalizableICUObject<UFieldPositionIterator *, ufieldpositer_close> FinalizableUFieldPositionIterator;
     typedef FinalizableICUObject<UCollator *, ucol_close> FinalizableUCollator;
 
-    template<typename ICUFunc>
-    static void RecyclerExecuteICUWithRetry(_In_ ICUFunc func, _In_ Recycler *recycler, _Out_writes_opt_(returnLen) char16 **ret, _Out_ int *returnLen, _In_ int firstTryLength = 8)
+    template<typename TExecutor>
+    static void EnsureBuffer(_In_ TExecutor executor, _In_ Recycler *recycler, _Outptr_result_buffer_(returnLength) char16 **ret, _Out_ int *returnLength, _In_ int firstTryLength = 8)
     {
-        bool success = ExecuteICUWithRetry<char16, ICUFunc>([&](int length)
+        UErrorCode status = U_ZERO_ERROR;
+        *ret = RecyclerNewArrayLeaf(recycler, char16, firstTryLength);
```

Comment:
Does this get cleaned up when you later reassign to it again on line 340? (Just want to make sure its not leaking.)

Reply:
Recycler is our GC, so it will get transiently cleaned if nothing references it anymore.